internal/runners/clean/run_lin_mac.go

```diff
@@ -25,6 +29,31 @@ func removeConfig(cfg configurable) error {
 	return os.RemoveAll(cfg.ConfigPath())
 }
 
-func removeInstall(installPath string) error {
-	return os.Remove(installPath)
+func removeInstall(_ configurable, installDir string) error {
+	stateInfo := appinfo.StateApp(installDir)
+	stateSvcInfo := appinfo.SvcApp(installDir)
+	stateTrayInfo := appinfo.TrayApp(installDir)
+
+	var aggErr error
+
+	for _, info := range []*appinfo.AppInfo{stateInfo, stateSvcInfo, stateTrayInfo} {
+		err := os.Remove(info.Exec())
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			aggErr = errs.Wrap(aggErr, "Could not remove %s: %v", info.Exec(), err)
+		}
+	}
+
+	appPath, err := installation.LauncherInstallPath()
+	if err != nil {
+		return errs.Wrap(aggErr, "Could not determine OS specific launcher install path")
+	}
```

Comment:
nit: we could move this to the top of the function and if an error is encountered here fail early without having removed anything from the user's system.